bgpd/bgp_open.c

```diff
@@ -594,11 +594,11 @@ bgp_capability_enhe (struct peer *peer, struct capability_header *hdr)
       /* RFC 5549 specifies use of this capability only for IPv4 AFI, with
        * the Nexthop AFI being IPv6. A future spec may introduce other
        * possibilities, so we ignore other values with a log. Also, only
-       * Unicast SAFI is currently supported (and expected).
+       * SAFI_UNICAST and SAFI_LABELED_UNICAST are currently supported (and expected).
        */
       nh_afi = afi_iana2int (pkt_nh_afi);
 
-      if (afi != AFI_IP || safi != SAFI_UNICAST || nh_afi != AFI_IP6)
+      if (afi != AFI_IP || nh_afi != AFI_IP6 || !(safi == SAFI_UNICAST || safi == SAFI_LABELED_UNICAST))
```

Comment:
Is there a reason to not also allow this for the other supported SAFIs (VPN, Encap, EVPN, ...)?